### PR TITLE
Azure iot/device provisioning service

### PR DESCRIPTION
**Pull request by montgomp**
## Description

Creating new autorest generated API for swagger spec:  [https://github.com/Azure/azure-rest-api-specs/pull/2093](https://github.com/Azure/azure-rest-api-specs/pull/2093)

---

This checklist is used to make sure that common guidelines for a pull request are followed.
- [X] Please add REST spec PR link to the SDK PR
- [X] **I have read the [contribution guidelines](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/.github/CONTRIBUTING.md).**
- [X] **The pull request does not introduce [breaking changes](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/Documentation/breaking-changes.md).**

### [General Guidelines](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/.github/CONTRIBUTING.md#general-guidelines)
- [X] Title of the pull request is clear and informative.
- [X] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/dev/documentation/cleaning-up-commits.md).

### [Testing Guidelines](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/.github/CONTRIBUTING.md#testing-guidelines)
- [X] Pull request includes test coverage for the included changes.

### [SDK Generation Guidelines](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/.github/CONTRIBUTING.md#sdk-generation-guidelines)
- [X] If an SDK is being regenerated based on a new swagger spec, a link to the pull request containing these swagger spec changes has been included above.
- [X] The generate.cmd file for the SDK has been updated with the version of AutoRest, as well as the commitid of your swagger spec or link to the swagger spec, used to generate the code.
- [X] The `*.csproj` and `AssemblyInfo.cs` files have been updated with the new version of the SDK.

### CODE REVIEW

**Review by shahabhijeet — CHANGES_REQUESTED** [review]

File: src/SDKs/ProvisioningServices/ProvisioningServices.Tests/ProvisioningServices.Tests.csproj
Comment (shahabhijeet) [inline src/SDKs/ProvisioningServices/ProvisioningServices.Tests/ProvisioningServices.Tests.csproj]: @montgomp remove commented code

File: src/SDKs/ProvisioningServices/ProvisioningServices.Tests/ProvisioningServices.Tests.csproj
Comment (shahabhijeet) [inline src/SDKs/ProvisioningServices/ProvisioningServices.Tests/ProvisioningServices.Tests.csproj]: @montgomp remove line 37 to 44.
all the references are being added to every test project

File: src/SDKs/ProvisioningServices/ProvisioningServices.Tests/ProvisioningServices.Tests.csproj
Comment (shahabhijeet) [inline src/SDKs/ProvisioningServices/ProvisioningServices.Tests/ProvisioningServices.Tests.csproj]: @montgomp remove line break

File: src/SDKs/ProvisioningServices/Management.ProvisioningServices/Microsoft.Azure.Management.ProvisioningServices.csproj
Comment (shahabhijeet) [inline src/SDKs/ProvisioningServices/Management.ProvisioningServices/Microsoft.Azure.Management.ProvisioningServices.csproj]: @montgomp if this is your first package, please release your first preview package as 0.9.0-preview
Once you are ready to go stable, release it as 1.0.0 (without preview)
make similar changes in AssemblyInfo.cs

**Review by shahabhijeet — CHANGES_REQUESTED** [review]
Can you do the following in the below order:
1)	Pull latest from upstream (if you have already done this good)
2)	Execute msbuild.exe build.proj from the repo root
a.	This will update the build tools in the current repo
3)	Generate your SDK
4)	Build you sdk at least one from command line of IDE
a.	Msbuild.exe build.proj /t:build /p:Scope=SDKs\ProvisioningServices
5)	Update the PR

File: src/SDKs/ProvisioningServices/ProvisioningServices.Tests/Helpers/Constants.cs
Comment (montgomp) [reply src/SDKs/ProvisioningServices/ProvisioningServices.Tests/Helpers/Constants.cs]: @shahabhijeet I don't follow what you're recommending here. This is similar to implementation in use in tests for Management.IoTHub

Do you have a link to an example of proposed changes?

File: src/SDKs/ProvisioningServices/Management.ProvisioningServices/Microsoft.Azure.Management.ProvisioningServices.csproj
Comment (shahabhijeet) [inline src/SDKs/ProvisioningServices/Management.ProvisioningServices/Microsoft.Azure.Management.ProvisioningServices.csproj]: @montgomp please remove lines 10-17
Add packageReleaseNotes
Model your project looking at the other SDK project files. you are missing 1 imports at the bottom of your project file.

File: src/SDKs/IotHub/AzSdk.RP.props
Comment (shahabhijeet) [inline src/SDKs/IotHub/AzSdk.RP.props]: @montgomp why was this file created? Please undo this change.

**Review by shahabhijeet — CHANGES_REQUESTED** [review]
Please generate using generate.cmd, that will create an additional meta data file needed for validating the PR.

File: src/SDKs/IotHub/AzSdk.RP.props
Comment (montgomp) [reply src/SDKs/IotHub/AzSdk.RP.props]: done

File: src/SDKs/ProvisioningServices/Management.ProvisioningServices/Microsoft.Azure.Management.ProvisioningServices.csproj
Comment (montgomp) [reply src/SDKs/ProvisioningServices/Management.ProvisioningServices/Microsoft.Azure.Management.ProvisioningServices.csproj]: added the import statement. Moved PackageReleaseNotes node into PropertyGroup that contains other package information

Comment (montgomp) [reply src/SDKs/ProvisioningServices/Management.ProvisioningServices/Microsoft.Azure.Management.ProvisioningServices.csproj]: Removing those lines will break compilation as it would remove a closing </PropertyGroup> and remove the TargetFrameworks node.

**Comment by shahabhijeet** [comment]
@montgomp you need AzSdk.RP.props file, please include it

File: src/SDKs/IotHub/AzSdk.RP.props
Comment (montgomp) [reply src/SDKs/IotHub/AzSdk.RP.props]: @shahabhijeet 7 days ago you told me to remove this file. Today (2018/01/09) you tell me I need the AzSdk.RP.props file. I am unable to fulfill both requests.

**Comment by montgomp** [comment]
@shahabhijeet the AzSdk.RP.props file that you asked me to recreate is the same one that you had asked me to remove earlier. It is being recreated automatically on each build and is in Microsoft.Azure.Management.IotHub assembly that is referenced by the test project. Please advise if you want the file included though no changes to IotHub are being performed as it is a separate assembly.

File: src/SDKs/ProvisioningServices/Management.ProvisioningServices/Microsoft.Azure.Management.ProvisioningServices.csproj
Comment (shahabhijeet) [inline src/SDKs/ProvisioningServices/Management.ProvisioningServices/Microsoft.Azure.Management.ProvisioningServices.csproj]: @montgomp any reason your service name in the swagger and your nuget package is not aligned.
For e.g. your nuget is called provisioning services, but your description uses Device Provisioning service and your API version in swagger is called just "devices"
This is just way too many names, your nuget name should be aligned with your service name.

**Review by shahabhijeet — CHANGES_REQUESTED** [review]

File: src/SDKs/ProvisioningServices/Management.ProvisioningServices/Microsoft.Azure.Management.ProvisioningServices.csproj
Comment (montgomp) [reply src/SDKs/ProvisioningServices/Management.ProvisioningServices/Microsoft.Azure.Management.ProvisioningServices.csproj]: I can change to Microsoft.Azure.Management.DeviceProvisioningService if that is the only thing holding up the PR. I cannot change the Microsoft.Devices in the swagger as that is not owned by my team. Please advise if this is acceptable as it is quite a large rework.

Comment (montgomp) [reply src/SDKs/ProvisioningServices/Management.ProvisioningServices/Microsoft.Azure.Management.ProvisioningServices.csproj]: The swagger changes you suggest would be a breaking change for python CLI that is already released.

Comment (shahabhijeet) [reply src/SDKs/ProvisioningServices/Management.ProvisioningServices/Microsoft.Azure.Management.ProvisioningServices.csproj]: The reason I have asked this is because, there is way for developers to search for a particular API version in nuget.org
So if your nuget is called ProvisioningService but your actual service is Devices, how will the developer know what to search for?
What is the official name of your service?

We cannot have different names for the service, it has to be consistent across.

Comment (montgomp) [reply src/SDKs/ProvisioningServices/Management.ProvisioningServices/Microsoft.Azure.Management.ProvisioningServices.csproj]: The service name is "[Microsoft Azure IoT Hub Device Provisioning Service](https://docs.microsoft.com/en-us/azure/iot-dps/)". I think that is quite the marketing mouthful which is why I'm more inclined to namespace as Microsoft.Azure.Management.DeviceProvisioningService. 

Again, I cannot change the Microsoft.Devices references due to 1) not being a contributor to the various projects if you want to go down that rabbit hole. 2) The number of production breaks that would be caused to ARM, portal, CLI, and existing PowerShell cmdlets for IoT Hub and Device Provisioning Service. 

Are you suggesting I move this assembly away from the Microsoft.Azure.Management.XXX hierarchy that other service SDKs appear to use?

**Comment by shahabhijeet** [comment]
@montgomp please reopen once you are ready to merge this.
This PR needs a fix in the spec, as soon as the spec PR is merged, you can either reopen or create a new PR.

**Pull request closed without merging**
